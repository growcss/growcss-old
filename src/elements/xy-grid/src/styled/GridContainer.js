// @flow
import styled from 'styled-components';

export const GridContainer = styled.div`
`;
