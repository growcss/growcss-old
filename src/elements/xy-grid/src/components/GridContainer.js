//@flow
import React, { Component } from 'react';
import remCalc from '@growcss/utils-remcalc';
import { Gutters as DefaultGutters } from './Gutters';
import { GridContainerElement } from '../styled/GridContainerElement';
import type { GuttersType } from '../types';

type GridContainerProps = {
  children?: any,
  layout?: string,
  width?: string | number,
  paddingGutter?: GuttersType | number | string,
};

export default class GridContainer extends Component<GridContainerProps>
{
  static defaultProps = {
    width: remCalc(1200),
    paddingGutter: DefaultGutters,
  };

  render() {
    const { layout, children, width, paddingGutter } = this.props;

    let maxWidth = width;
    let gutter = paddingGutter;

    if (layout === 'fluid' || layout === 'full') {
      maxWidth = '100%';
    }

    if (layout === 'full') {
      gutter = '0';
    }

    return (
      <GridContainerElement maxWidth={maxWidth} paddingGutter={gutter} layout={layout}>
        {children}
      </GridContainerElement>
    );
  }
}
