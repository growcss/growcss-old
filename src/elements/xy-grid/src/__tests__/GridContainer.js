//@flow
import React from 'react';
import GridContainer from '../components/GridContainer';

test('Link changes the class when hovered', () => {
  const wrapper = shallow(<GridContainer>test</GridContainer>);

  expect(wrapper).toMatchSnapshot();
});
