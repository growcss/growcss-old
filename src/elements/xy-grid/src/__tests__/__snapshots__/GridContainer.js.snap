// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if the container content is of the full width of the available space. 1`] = `
<GridContainerElement
  layout="full"
  maxWidth="100%"
  paddingGutter="0"
>
  test
</GridContainerElement>
`;

exports[`if the container is centered and has a max-width of 100%. 1`] = `
<GridContainerElement
  layout="fluid"
  maxWidth="100%"
  paddingGutter={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
>
  test
</GridContainerElement>
`;

exports[`if the container is centered and have a max-width of 75rem. 1`] = `
<GridContainerElement
  maxWidth="75rem"
  paddingGutter={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
>
  test
</GridContainerElement>
`;
